datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/util/OwnerUtils.java

```diff
@@ -150,76 +136,79 @@ private static void addOwner(
             : com.linkedin.common.OwnershipType.CUSTOM;
 
     newOwner.setType(gmsType);
-    newOwner.setTypeUrn(ownershipUrn);
+    newOwner.setTypeUrn(ownershipTypeUrn);
     newOwner.setSource(new OwnershipSource().setType(OwnershipSourceType.MANUAL));
     newOwner.setOwner(ownerUrn);
     ownerArray.add(newOwner);
     ownershipAspect.setOwners(ownerArray);
   }
 
+  private static void removeExistingOwnerIfExists(
+      OwnerArray ownerArray, Urn ownerUrn, Urn ownershipTypeUrn) {
+    ownerArray.removeIf(
+        owner -> {
+          // Remove old ownership if it exists (check ownerUrn + type (entity & deprecated type))
+          return isEqual(owner, ownerUrn, ownershipTypeUrn);
+        });
+  }
+
+  public static boolean isEqual(Owner owner, Urn ownerUrn, Urn ownershipTypeUrn) {
+    if (!owner.getOwner().equals(ownerUrn)) {
+      return false;
+    }
+    if (owner.getTypeUrn() != null) {
+      return owner.getTypeUrn().equals(ownershipTypeUrn);
+    }
+    if (ownershipTypeUrn == null) {
+      return false;
+    }
```

Comment:
should this return true in this situation? if there's no ownershipTypeUrn then we should default to remove if `owner.getOwner().equals(ownerUrn)`? at least that's how we had it before it looks like to me